frontend/src/v5/services/realtime/realtime.service.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { io, Socket } from 'socket.io-client';
+
+const CHAT_SERVER = 'http://local.3drepo.io:3000';
+
+interface IRoomType {
+	teamspace: string;
+	model?: string;
+	project: string;
+}
+
+let socket:Socket = null;
+const roomsJoined:Record<string, number> = {};
+
+const roomTypeToId = (roomType:IRoomType) => `${roomType.teamspace}.${roomType.project}.${roomType.model}`;
+
+export const initializeSocket = () => {
+	socket = io(CHAT_SERVER, {
```

Comment:
we need to handle reconnection by hooking into the connect event, [like v4](https://github.com/3drepo/3drepo.io/blob/staging/frontend/src/v4/modules/chat/chat.sagas.ts#L41)

Reply:
message event needs to be listened to for logout events https://github.com/3drepo/3drepo.io/blob/staging/frontend/src/v4/modules/chat/chat.sagas.ts#L43

Reply:
Another thing is that the backend sends back error messages like this: https://github.com/3drepo/3drepo.io/blob/master/backend/src/v4/services/chat.js#L129


But i'm thinking now whether I should change it in v5 to be more like the logout events? (so it's a direct message with the details)

Otherwise you'd have to hook onto every possible error event to get any error handling.